src/test/java/peoplesoft/model/AddressBookTest.java

```diff
@@ -57,25 +58,47 @@ public void resetData_withDuplicatePersons_throwsDuplicatePersonException() {
 
     @Test
     public void hasPerson_nullPerson_throwsNullPointerException() {
-        assertThrows(NullPointerException.class, () -> addressBook.hasPerson(null));
+        assertThrows(NullPointerException.class, () -> addressBook.hasPerson((Person) null));
```

Comment:
wouldn't be a bad idea to check for both `(ID) null` and `(Person) null`

Reply:
same for the other tests, might as well test the other codepath(s) if it only takes one more line of code

Reply:
oh ok, I see that you do test for those below, please disregard